builder/oracle/bmcs/builder.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2017 Oracle America, Inc.
+// The contents of this file are subject to the Mozilla Public License Version
+// 2.0 (the "License"); you may not use this file except in compliance with the
+// License. If a copy of the MPL was not distributed with this file, You can
+// obtain one at http://mozilla.org/MPL/2.0/
+
+// Package bmcs contains a packer.Builder implementation that builds Oracle
+// Bare Metal Cloud Services (BMCS) images.
+package bmcs
+
+import (
+	"fmt"
+	"log"
+
+	client "github.com/hashicorp/packer/builder/oracle/bmcs/client"
+	"github.com/hashicorp/packer/common"
+	"github.com/hashicorp/packer/helper/communicator"
+	"github.com/hashicorp/packer/packer"
+	"github.com/mitchellh/multistep"
+)
+
+// BuilderId uniquely identifies the builder
+const BuilderId = "packer.oracle.bmcs"
+
+// BMCS API version
+const bmcsAPIVersion = "20160918"
+
+// Builder is a builder implementation that creates Oracle BMCS custom images.
+type Builder struct {
+	config *Config
+	runner multistep.Runner
+}
+
+func (b *Builder) Prepare(rawConfig ...interface{}) ([]string, error) {
+	config, err := NewConfig(rawConfig...)
+	if err != nil {
+		return nil, err
+	}
+	b.config = config
+
+	return nil, nil
+}
+
+func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packer.Artifact, error) {
+	driver, err := NewDriverBMCS(b.config)
+	if err != nil {
+		return nil, err
+	}
+
+	// Populate the state bag
+	state := new(multistep.BasicStateBag)
+	state.Put("config", b.config)
+	state.Put("driver", driver)
+	state.Put("hook", hook)
+	state.Put("ui", ui)
+
+	// Build the steps
+	steps := []multistep.Step{
+		&stepKeyPair{
+			Debug:          b.config.PackerDebug,
+			DebugKeyPath:   fmt.Sprintf("bmcs_%s.pem", b.config.PackerBuildName),
+			PrivateKeyFile: b.config.Comm.SSHPrivateKey,
+		},
+		&stepCreateInstance{},
+		&stepInstanceInfo{},
+		&communicator.StepConnect{
+			Config: &b.config.Comm,
+			Host:   commHost,
+			SSHConfig: SSHConfig(
+				b.config.Comm.SSHUsername,
+				b.config.Comm.SSHPassword),
+		},
+		&common.StepProvision{},
+		&stepImage{},
+	}
+
+	// Run the steps
+	b.runner = common.NewRunner(steps, b.config.PackerConfig, ui)
```

Comment:
This is the wrong way of doing it. Check how it's done in [virtualbox-iso](https://github.com/hashicorp/packer/blob/master/builder/virtualbox/iso/builder.go#L296).

Reply:
Hey @rickard-von-essen do you see this as a critical fix? I'm very keen to get this into the next release since we've had the PR open for 7 months now. If possible we would merge this and submit a fix straight after the first release. Let me know what you think.

Reply:
It is a very simple fix. And without it `packer build -on-error=...` won't work. So it would be good.

Reply:
Ok no problem. We'll fix. Thanks for the info 👍

Reply:
This should be resolved by dbbe866